### PR TITLE
Fix error in logging: could not concat bytes and string

### PR DESCRIPTION
**Pull request by mniak**
When a media was received, an error was thrown because of this logging issue.
This is my proposed solution, that works for me, but I don't know if it satisfies

### CODE REVIEW

**Comment by mniak** [comment]
#2016

This pull request fixes the receipt of the field `media_key` and this way you are already able to echo the images. For this, in the `EchoLayer` you will need to add the `mediaKey` parameter like this: 
```
    def send_image_echo(self, entity):
        echo = ImageDownloadableMediaMessageProtocolEntity(
            entity.getMimeType(), entity.fileHash, entity.url,
            entity.ip,
            entity.size, entity.fileName, entity.encoding,
            entity.width, entity.height,
            entity.getCaption(), entity.mediaKey,
            to=entity.getFrom(), preview=entity.getPreview())

        print("Echoing image")
        self.toLower(echo)
```

**Comment by bombadiltom** [comment]
Any other pull requests needed here?
If i fetch this one into master, i get the following error at login: 
   val = ''.join(['%' + hex(b)[2:] for b in val])
TypeError: hex() argument can't be converted to hex

**Comment by mniak** [comment]
@bombadiltom are you using python2.7? I was using 3 and this error was not thrown to me. 
Are you doing something different at login?

I will test in 2.7

**Comment by bombadiltom** [comment]
@mniak Python 2.7.9, yowsup-cli demos -c /etc/yowsup.config -y, then /L

**Comment by bombadiltom** [comment]
@mniak: Python3:
[offline]:/L
Auth: Logged in!
[connected]:/image send 12345 /space/doorbird/lastmotion.jpg
[...]
File "/usr/local/lib/python3.4/dist-packages/yowsup2-2.5.2-py3.4.egg/yowsup/layers/__init__.py", line 163, in processIqRegistry
    errorClbk(protocolTreeNode, originalIq)
  File "/usr/local/lib/python3.4/dist-packages/yowsup2-2.5.2-py3.4.egg/yowsup/layers/axolotl/layer_base.py", line 81, in onError
    errorClbk(errorNode, getKeysEntity)
TypeError: <lambda>() takes 0 positional arguments but 2 were given

BTW: Did you test your pull request with pull request 555 already? What's the best way to easilly merge them together?

**Comment by langioletto** [comment]
I applied the changes and I launched the layer.py again

`File "/usr/local/lib/python2.7/dist-packages/yowsup2-2.5.2-py2.7.egg/yowsup/layers/axolotl/layer_control.py", line 34, in send
self.toLower(node)

File "/usr/local/lib/python2.7/dist-packages/yowsup2-2.5.2-py2.7.egg/yowsup/layers/init.py", line 84, in toLower
self.__lower.send(data)

File "/usr/local/lib/python2.7/dist-packages/yowsup2-2.5.2-py2.7.egg/yowsup/layers/logger/layer.py", line 8, in send
logger.debug("tx:\n%s" % ldata)

File "/usr/local/lib/python2.7/dist-packages/yowsup2-2.5.2-py2.7.egg/yowsup/structs/protocoltreenode.py", line 90, in str
return self.toString()

File "/usr/local/lib/python2.7/dist-packages/yowsup2-2.5.2-py2.7.egg/yowsup/structs/protocoltreenode.py", line 56, in toString
val = ''.join(['%' + hex(b)[2:] for b in val])
TypeError: hex() argument can't be converted to hex
`

The issue is generated with changes to: protocoltreenode.py

File: yowsup/structs/protocoltreenode.py [hunk 1]
```diff
@@ -49,10 +49,12 @@ def __hash__(self):
     def toString(self):
         out = "<"+self.tag
         if self.attributes is not None:
-            for key,val in self.attributes.items():
+            for key, val in self.attributes.items():
                 if val is None:
                     raise ValueError("value is none for attr %s" % key)
-                out+= " "+key+'="'+val+'"'
+                if type(val) == bytes:
+                    val = ''.join(['%' + hex(b)[2:] for b in val])
```
Comment (leogaletti) [inline yowsup/structs/protocoltreenode.py]: error in this line

**Comment by langioletto** [comment]
News?

With python2.7 these changes do not work, can you confirm?

**Comment by AragurDEV** [comment]
So the image echoing is working. But I can't get it working to send a image from filePath
Exception thrown:
'https://longurl' has type <class 'str'>, but expected one of: ((<class 'bytes'>,),)

If I convert the URL via .encode i get following thrown:
None has type <class 'NoneType'>, but expected one of: ((<class 'bytes'>,),)
(A log of the converted value returns b https://longurl)
[URL was shortened to improve readability]

**Comment by luidna** [comment]
Removed the logging modification from pull request.
Only media echo fixes included

**Comment by mniak** [comment]
As no one is accepting this, I will then close this PR.

**Pull request closed without merging**